src/modules/referral/services/queries.ts

```diff
@@ -76,3 +76,27 @@ export const getActiveRefereesCountQuery = () => {
     where temp.r = 1 and temp."referralAddress" = $1;
   `;
 };
+
+export const updateStickyReferralAddresses = () => {
```

Comment:
How often does this update get run? On every addition of a new element? Does it run over all previous elements and update them too?

Reply:
@mrice32 

It runs whenever a new deposit is detected

> Does it run over all previous elements and update them too?

Yes! It performs a join SQL operation to update the referral address of all the deposits performed by the same depositor address

Reply:
Interesting!

Pretty wild that you can do all this in SQL.